flycheck.el

```diff
@@ -1110,6 +1110,22 @@ Return the path of the directory"
     (push tempdir flycheck-temporaries)
     tempdir))
 
+(defvar flycheck--forced-suffix nil
+  "List of forced filename suffixes.
+
+Dynamic variable.  If non-nil, ensure generated file names have
+one of the suffixes in this list.  Activated using
+a (force-suffix ...) clause in `flycheck-substitute-argument'.")
```

Comment:
I'm not a huge fan of dynamic variables; though I'm not sure whether there's a uch better alternative yet.

Reply:
It could be made a property of the checker, though that would give less fine-grained control over which filenames are affected (in case we want to generate more than one filename in an invocation).

Reply:
Indeed, a property of the checker might work.  I don't think we ever generate more than one filename — can we think of a case in which we would want to?